### PR TITLE
fix Quaternion.slerp for small angles

### PR DESCRIPTION
**Pull request by mbredif**
Previous implementation relied on average quaternions (t=0.5) for small angles.
Now it uses the same approach as Quaternion.slerpFlat (same threshold and lerp interpolation).

This PR tries to address #14258 and iTowns/itowns#774

### CODE REVIEW

**Comment by WestLangley** [comment]
`slerp()` and `slerpFlat()` should not be using different algorithms. We will have to fix that.

Be that as it may, perhaps you should add `this.normalize()` after lerping; `lerp()` will not maintain unit length.

In fact, can you please check the consequence of failing to normalize in your app?

File: src/math/Quaternion.js [hunk 1]
```diff
@@ -533,19 +533,21 @@ Object.assign( Quaternion.prototype, {
 
 		}
 
-		var sinHalfTheta = Math.sqrt( 1.0 - cosHalfTheta * cosHalfTheta );
+		var sqrSinHalfTheta = 1.0 - cosHalfTheta * cosHalfTheta;
 
-		if ( Math.abs( sinHalfTheta ) < 0.001 ) {
+		if ( sqrSinHalfTheta <= Number.EPSILON ) {
```
Comment (mrdoob) [inline src/math/Quaternion.js]: `Math.abs()` is no longer needed?

Comment (mbredif) [reply src/math/Quaternion.js]: no : a square is always non negative !
do you prefer a update of this PR with a call to normalize() or a simply implementing slerp with a subcall to slerpFlat ?

Comment (WestLangley) [reply src/math/Quaternion.js]: If `cosHalfTheta` = - 1.00001 due to roundoff then `sqrSinHalfTheta` could be negative.

Comment (mbredif) [reply src/math/Quaternion.js]: I agree, but that is no longer an issue : negative `sqrSinHalfTheta` corresponds to cases where the cos is round off errors away from 1 or -1, which corresponds to tiny values of the sin, for which lerp makes sense. (it WAS however an issue as the sqrt was not guarded against negative values and would output NaN values...)

**Comment by WestLangley** [comment]
Thank you @mbredif.

@mrdoob This can be merged.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**